flixel/system/FlxBuffer.hx

```diff
@@ -0,0 +1,478 @@
+package flixel.system;
+
+import haxe.macro.TypeTools;
+#if macro
+import haxe.macro.Type;
+import haxe.macro.Expr;
+import haxe.macro.Context;
+
+using haxe.macro.Tools;
+
+class BufferMacro
+{
+	public static function build(isVector:Bool, includeGetters = true):ComplexType
+	{
+		final local = Context.getLocalType();
+		switch local {
+			// Extract the type parameter
+			case TInst(local, [type]):
+				return buildBuffer(getFields(type, includeGetters), type, isVector);
+			default:
+				throw "Expected TInst";
```

Comment:
I think using `Context.error()` would be better than throwing since it shows the error at the location where TInst was expected